src/components/KeyblockDetailsPanel.vue

```diff
@@ -0,0 +1,220 @@
+              <span class="keyblock-details-panel__hash">
+                {{ keyblockDetails.beneficiary }}
+              </span>
+              <span class="keyblock-details-panel__hash-ellipse">
+                {{ formatEllipseHash(keyblockDetails.beneficiary) }}
+              </span>
+            </app-link>
+          </td>
+        </tr>
+        <tr class="keyblock-details-panel__row">
+          <th class="keyblock-details-panel__table-header">
+            Beneficiary Reward
+          </th>
+          <td class="keyblock-details-panel__data">
+            {{ formatAePrice(keyblockDetails.block_reward, null) }}
+          </td>
+        </tr>
+        <tr class="keyblock-details-panel__row">
+          <th class="keyblock-details-panel__table-header">
+            BRI Reward
+          </th>
+          <td class="keyblock-details-panel__data">
+            {{ formatAePrice(keyblockDetails.dev_reward, null) }}
+          </td>
+        </tr>
+        <tr class="keyblock-details-panel__row">
+          <th class="keyblock-details-panel__table-header">
+            Microblocks Count
+          </th>
+          <td class="keyblock-details-panel__data">
+            {{ formatNumber(keyblockDetails.micro_blocks_count) }}
+          </td>
+        </tr>
+        <tr class="keyblock-details-panel__row">
+          <th class="keyblock-details-panel__table-header">
+            Transactions Count
+          </th>
+          <td class="keyblock-details-panel__data">
+            {{ formatNumber(keyblockDetails.transactions_count) }}
+          </td>
+        </tr>
+        <tr class="keyblock-details-panel__row">
+          <th class="keyblock-details-panel__table-header">
+            API Links
+          </th>
+          <td class="keyblock-details-panel__data">
+            <div class="keyblock-details-panel__container">
+              <app-link
+                :to="keyblockNodeUrl"
+                class="keyblock-details-panel__link">
+                <app-icon
+                  name="file-cloud"
+                  :size="22"/>
+                Node
+              </app-link>
+
+              <app-link
+                :to="keyblockMiddlewareUrl"
+                class="keyblock-details-panel__link">
+                <app-icon
+                  name="file-cloud"
+                  :size="22"/>
+                Middleware
+              </app-link>
+            </div>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </app-panel>
+</template>
+
+<script setup>
+import AppPanel from '@/components/AppPanel'
+import CopyChip from '@/components/CopyChip'
+import AppIcon from '@/components/AppIcon'
+import AppLink from '@/components/AppLink'
+import { formatAePrice, formatEllipseHash, formatNumber } from '@/utils/format'
+
+const { NODE_URL, MIDDLEWARE_URL } = useRuntimeConfig().public
+
+const props = defineProps({
+  keyblockDetails: {
+    type: Object,
+    required: true,
+  },
+})
+
+const keyblockNodeUrl = computed(() =>
```

Comment:
One more thing to consider - Would this find a better home in adapters? It is also valid for more entities. Maybe it is worth refactoring issue?

Reply:
Interesting idea. I like keeping it in one place along with other data transformation but on the other hand it's used only in one place per entity type while the adapters are generally reusable (e.g. adaptKeyblock is used in multiple places). I don't have a strong opinion. @michele-franchi wdyt?

Reply:
Yes that's my suggestion. My intention to create adapters was to move as most logic from templates and create the layer that would move mapping from MDW aside